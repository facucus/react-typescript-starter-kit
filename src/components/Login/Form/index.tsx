import React from "react";

interface IFormProps {
  onClick: () => void;
}

const Form: React.SFC<IFormProps> = props => {
  return (
    <form>
      <div>
        <label htmlFor="username">Username</label>
        <input id="username" placeholder="username" name="username" />
      </div>

      <div>
        <label htmlFor="password">Password</label>
        <input
          id="password"
          type="password"
          placeholder="password"
          name="password"
        />
      </div>

      <button type="button" onClick={props.onClick}>
        Click to Login
      </button>
    </form>
  );
};

export default Form;
