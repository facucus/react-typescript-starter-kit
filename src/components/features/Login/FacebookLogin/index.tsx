import FacebookLogin from "react-facebook-login";
import React, { PureComponent } from "react";

interface ILoginState {
  isLogin: boolean;
  userID?: string;
  name?: string;
  email?: string;
  picture?: string;
}

interface IDataResponse {
  height: number;
  is_silhouette: boolean;
  url: string;
  width: number;
}

interface IPictureData {
  data: IDataResponse;
}

interface IFacebookResponse {
  accessToken: string;
  data_access_expiration_time: number;
  expiresIn: number;
  id: string;
  name: string;
  reauthorize_required_in: number;
  signedRequest: string;
  userID: string;
  email: string;
  picture: IPictureData;
}

export default class Index extends PureComponent<{}, ILoginState> {
  public state = {
    isLogin: false,
    userID: "",
    name: "",
    email: "",
    picture: ""
  };

  public responseFacebook = (response: IFacebookResponse) => {
    this.setState({
      isLogin: true,
      userID: response.userID,
      name: response.name,
      email: response.email,
      picture: response.picture.data.url
    });
  };

  public render() {
    let fbContent;

    if (!process.env.REACT_APP_FACEBOOK_ID) {
      return null;
    }

    if (this.state.isLogin) {
      fbContent = (
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            justifyContent: "space-around",
            backgroundColor: "lightGray",
            width: "350px",
            height: "125px"
          }}
        >
          <img alt={this.state.name} src={this.state.picture} />
          <h2>Welcome {this.state.name}</h2>
          <p>Email: {this.state.email}</p>
        </div>
      );
    } else {
      fbContent = (
        <FacebookLogin
          appId={process.env.REACT_APP_FACEBOOK_ID}
          autoLoad={true}
          fields="name,email,picture"
          callback={this.responseFacebook}
          textButton="Zenfolio login with Facebook"
        />
      );
    }

    return <div>{fbContent}</div>;
  }
}
