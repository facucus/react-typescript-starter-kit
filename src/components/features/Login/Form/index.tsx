import React from "react";
import Button from "../../../ui/Button";
import styles from "./form.module.scss";

interface IFormProps {
  username?: string;
  password?: string;
  onClick: () => void;
  onChangeInput: (evt: React.ChangeEvent<HTMLInputElement>) => void;
}

interface IEventTarget {
  addEventListener(
    type: string,
    listener: EventListenerOrEventListenerObject,
    useCapture?: boolean
  ): void;
  dispatchEvent(evt: Event): boolean;
  removeEventListener(
    type: string,
    listener: EventListenerOrEventListenerObject,
    useCapture?: boolean
  ): void;
}

class Form extends React.Component<IFormProps> {
  private handleSubmit = (evt: React.FormEvent<IEventTarget>): void => {
    evt.preventDefault();
    this.props.onClick();
  };

  public render() {
    return (
      <form onSubmit={this.handleSubmit} className={styles.container}>
        <div>
          <label htmlFor="username">Username</label>
          <input
            id="username"
            placeholder="username"
            name="username"
            onChange={this.props.onChangeInput}
            value={this.props.username}
          />
        </div>

        <div>
          <label htmlFor="password">Password</label>
          <input
            id="password"
            type="password"
            placeholder="password"
            name="password"
            onChange={this.props.onChangeInput}
            value={this.props.password}
          />
        </div>
        <Button type="submit" onClick={this.handleSubmit}>
          Click to Login
        </Button>
      </form>
    );
  }
}

export default Form;
