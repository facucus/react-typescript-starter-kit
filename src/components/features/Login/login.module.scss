@import "../../../styles/styleguide.scss";

.container {
  display: flex;
  flex-direction: column;
  padding: 50px;
}

.title {
  color: red;
}
