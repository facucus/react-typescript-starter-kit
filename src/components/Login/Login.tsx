import React from "react";
import { Redirect, RouteComponentProps } from "react-router-dom";

import styles from "./login.module.scss";
import Form from "./Form";

interface ILoginProps extends RouteComponentProps {
  loggedIn: boolean;
  error?: string;
  onFetchLogin: (userData: { username: string; password: string }) => void;
}

interface ILoginState {
  username?: string;
  password?: string;
}

class Login extends React.Component<ILoginProps, ILoginState> {
  public state = {
    username: "",
    password: ""
  };

  public submitForm = () => {
    const { username, password } = this.state;
    this.props.onFetchLogin({ username, password });
  };

  public onChangeInput = (evt: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ [evt.target.id]: evt.target.value });
  };

  public render() {
    if (this.props.loggedIn) {
      const { from } = this.props.location.state || { from: { pathname: "/" } };
      return <Redirect to={"/"} />;
    }
    return (
      <div>
        <h1 className={styles.title}>Login</h1>
        <Form
          password={this.state.password}
          username={this.state.username}
          onChangeInput={this.onChangeInput}
          onClick={this.submitForm}
        />
        {this.props.error && (
          <p data-testid="login-error-msg">{this.props.error}</p>
        )}
      </div>
    );
  }
}

export default Login;
