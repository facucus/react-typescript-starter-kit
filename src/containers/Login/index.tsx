import { Dispatch } from "react";
import { IAppState } from "../../store";
import * as actions from "../../actions/session";
import { connect } from "react-redux";
import Login from "../../components/Login/Login";
import { ThunkDispatch } from "redux-thunk";

function mapStateToProps({ session }: IAppState) {
  return {
    loggedIn: session.loggedIn,
    error: session.error
  };
}

function mapDispatchToProps(dispatch: ThunkDispatch<any, any, any>) {
  return {
    onChangeLoggedIn: (newValue: boolean) =>
      dispatch(actions.changeLogin(newValue)),
    onFetchLogin: (userData: { username: string; password: string }) =>
      dispatch(actions.fetchLogin(userData))
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
