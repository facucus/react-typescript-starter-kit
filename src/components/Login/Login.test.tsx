import "jest-dom/extend-expect";
import "react-testing-library/cleanup-after-each";
import "jest-axe/extend-expect";
import React from "react";
import { render } from "react-testing-library";
import Login from "./Login";

describe("Login Page", () => {
  const props = {
    loggedIn: false,
    onChangeLoggedIn: jest.fn()
  };

  test("It should have a Login title <h1>", () => {
    const { container } = render(<Login {...props} />);

    const title = container.querySelector("h1");

    expect(title).toHaveTextContent("Login");
  });

  test("Match Snapshot", () => {
    const { container } = render(<Login {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
