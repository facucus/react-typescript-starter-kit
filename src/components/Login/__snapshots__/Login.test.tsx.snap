// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Match Snapshot 1`] = `
<div>
  <h1
    class="title"
  >
    Login
  </h1>
  <form>
    <div>
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        placeholder="username"
      />
    </div>
    <div>
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        type="password"
      />
    </div>
    <button
      type="button"
    >
      Click to Login
    </button>
  </form>
</div>
`;
